modules/elements-color-picker/assets/js/editor/commands/show-swatches.js

```diff
@@ -0,0 +1,283 @@
+import CommandBase from 'elementor-api/modules/command-base';
+import { rgbToHex } from 'elementor/core/app/assets/js/utils/utils';
+
+export class ShowSwatches extends CommandBase {
+	constructor( args ) {
+		super( args );
+
+		this.colors = {};
+		this.pickerClass = 'e-element-color-picker';
+		this.pickerSelector = '.' + this.pickerClass;
+		this.container = null;
+		this.backgroundImages = [];
+	}
+
+	/**
+	 * Validate the command arguments.
+	 *
+	 * @param args
+	 */
+	validateArgs( args ) {
+		this.requireArgument( 'event', args );
+	}
+
+	/**
+	 * Execute the command.
+	 *
+	 * @param args
+	 */
+	apply( args ) {
+		const { event: e } = args;
+		const id = e.currentTarget.dataset.id;
+
+		// Calculate swatch location.
+		const rect = e.currentTarget.getBoundingClientRect(),
+			x = Math.round( e.clientX - rect.left ) + 'px',
+			y = Math.round( e.clientY - rect.top ) + 'px';
+
+		// Don't pick colors from the current widget.
+		if ( id === this.component.currentPicker.container.id ) {
```

Comment:
1. Pretty useless, you will get the same color that you're currently seeing.
2. It causes issues when entering preview on hover, because the element is re-rendered, which then removes the colors palette.

Reply:
Done.